src/onegov/core/utils.py

```diff
@@ -416,7 +417,16 @@ def linkify(text: str, escape: bool = True) -> str:
     if not text:
         return text
 
-    linkified = linkify_phone(bleach.linkify(text, parse_email=True))
+    long_top_level_domains = ['.agency']
+
+    # bleach.linkify supports only a fairly limited amount of tlds
+    if any(domain in text for domain in long_top_level_domains):
+        if '@' in text:
+            linkified = str(Markup(f'<a href="mailto:{text}">{text}</a>'))
```

Comment:
This is not actually safe, you are bypassing the HTML escaping by using a f-string, you need to use `.format` on the `Markup` i.e. `Markup('<a href="mailto:{text}">{text}</a>').format(text=text)`

The purpose of the `Markup` class is to mark text that already contains markup that you validated and to mix in safely with regular strings. so `Markup` + `str` is the same as `Markup + escape(str)`

Reply:
I see. Fixed now.